### DISCUSSION TITLE
Github pages only loading README or blank page for React website

### DISCUSSION THREAD

**Discussion by danyabaron** [community/community#40288] [Other Feature Feedback, Questions, & Ideas]
### Github Pages loading blank page or README for React personal site

I have followed the steps from (https://github.com/gitname/react-gh-pages) to create a personal website at (https://danyabaron.github.io/), however, I have been troubleshooting for the last week to get it to deploy properly to the site address. It has just been loading my README file for the past week. 

I have two branches, main and gh-pages. gh-pages is not updated with my main code, so in settings, I have selected to load the site from the main branch and root directory. I also have a build deploy script in my package.json to build the site from the main branch and I set the homepage to be https://danyabaron.github.io/.  I have tried to merge my code from main branch into gh-pages branch but have had lots of issues in doing so. I read here that if the index.html is in another file, to move it to the root directory. I moved the index.html file into the root directory and the page is now blank.

I would greatly appreciate some help in deploying my website properly and getting it to load to https://danyabaron.github.io/. My repo is here: https://github.com/danyabaron/danyabaron.github.io

**Comment by airtower-luna** [comment]
> I have selected to load the site from the main branch and root directory. I also have a build deploy script in my package.json to build the site from the main branch

GitHub Pages has nothing built-in to handle a React site, by default it only runs your file through [Jekyll](https://jekyllrb.com/), and deploys the result. `package.json` is not a file treated specially by Jekyll, so it'll be deployed as-is (check https://danyabaron.github.io/package.json).

> I moved the index.html file into the root directory and the page is now blank.

That's because the body of your `index.html` file contains only an empty `<div>` element.

I assume you'll want to run a React build and deploy the result. In principle, you have two ways to do that:

1. Use a [custom deployment workflow](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow) that does the build and deploys the result. This is the newer, generally recommended method. There are [some starter workflows](https://github.com/actions/starter-workflows/tree/main/pages) you could use as examples.
2. Do the build manually, and commit the result to a separate branch (e.g. `gh-pages`) and deploy from there.

Either way make sure the build creates an `index.html` file that actually contains your application.

By the way, none of the commits in that repository are recognized as yours (as in, connected to your account). You might want to either fix the author email you use when committing, or if it's correct add it to your account. :wink:

**Reply by danyabaron** [reply]
Thanks so much for your answer. I am currently looking into the custom deployment workflow, but am a bit confused by them. For the build manually, do you mean to create a build locally and then commit it to gh-pages? 

lol, I had no idea that the commits were not recognized as mine I have no idea how that happened, how do you fix the author email? I was looking in my settings but could not find the option to do so

**Reply by danyabaron** [reply]
also so the build should be making a new index.html file? I do have a div named root in there, but in another folder called 'src' is where I have all the react code filling in that root div, but it seems that the deployment is not recognizing that src folder

**Reply by airtower-luna** [reply]
> I am currently looking into the custom deployment workflow, but am a bit confused by them.

It's a regular [Actions workflow](https://docs.github.com/en/actions/quickstart), if you haven't worked with CI before there's a bit of a learning curve. But having it all automated is well worth it in my opinion. :wink: 

> For the build manually, do you mean to create a build locally and then commit it to gh-pages?

Yes, exactly. With the files exactly as you want them to be deployed.

> I had no idea that the commits were not recognized as mine I have no idea how that happened, how do you fix the author email?

You can check the currently used address with `git config user.email`. If that's the address you want to use just add it to your account, if you want to change it use `git config --global user.email YOUR_EMAIL_HERE`. See also: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

**Reply by danyabaron** [reply]
Okay thank you so much! I will take a look at the automated builds. I realize I may have screwed up my gh-pages branch as well..

and thank you for the tip regarding the user address!!
